templates/common/cloud-platform-alt-dns/files/coredns.yaml

```diff
@@ -0,0 +1,113 @@
+mode: 0644
+path: {{ if gt (len (cloudPlatformAPIIntLoadBalancerIPs .)) 0 -}} "/etc/kubernetes/manifests/coredns.yaml" {{ else }} "/etc/kubernetes/disabled-manifests/coredns.yaml" {{ end }}
+contents:
+  inline: |
+    kind: Pod
+    apiVersion: v1
+    metadata:
+      name: coredns
+      namespace: openshift-{{ cloudPlatformShortName . }}-infra
```

Comment:
I would recommend not using a platform-specific namespace. If I had it to do over again I'd have used openshift-on-prem-infra (or something like that) for all of the on-prem pods. The separate names just clutter up the namespaces and don't add any benefit.

Also, whatever you do use for the namespace it will need to be added to https://github.com/openshift/machine-config-operator/blob/master/install/0000_80_machine-config-operator_00_namespace.yaml (unless I missed where it was being added).

Reply:
That is great advice. I am using the name "openshift-cloud-platform-infra". Let me know if that works.